### PR TITLE
Update Version.js to allow user to define a proxy server

### PR DESCRIPTION
**Pull request by a632079**
In China, Github might be blocked by the GFW.
It means that nodebb can't fetch data from Github API.
So, I add a config option to allow user to define their own api.

### CODE REVIEW

**Comment by pitaj** [comment]
Have you tested trying to provide your own API? I'd prefer allowing use of a PROXY environmental variable instead.

**Comment by a632079** [comment]
I quite approve of using proxy as recommended method. But Http or Socket proxy is not a good choice in China. As you know, the policy is not allowed to use unofficial VPN/Proxy. It means it is illegal to use a socket/http proxy. (Because they don't know what you will visit over the proxy). 

I had built a github api proxy (https://apis.nodebb-cn.org) and a nodebb api proxy (https://packages.nodebb-cn.org) via a singapore VPS which was provided by AlibabaCloud. I think it is a easy way to let user get speeder api request. Thanks

**Comment by pitaj** [comment]
I'd still prefer a more general variable like:

    nconf.any('GITHUB_API', 'github_api')

Which would provide the base path to the GitHub API: `https://api.github.com` or some alternative.


We could do the same for the NodeBB Packages API.

**Comment by a632079** [comment]
I found `nconf.get('registry')` was be defined as mirror option. Need me edit it, replacing with which like `nconf.get('nodebb_package_api')`?

**Comment by pitaj** [comment]
No you don't need to change that. Handy that is already exists though.

File: src/admin/versions.js [hunk 1]
```diff
@@ -20,7 +21,7 @@ function getLatestVersion(callback) {
 		headers['If-Modified-Since'] = versionCacheLastModified;
 	}
 
-	request('https://api.github.com/repos/NodeBB/NodeBB/tags', {
+	request(nconf.get('github_api') || 'https://api.github.com/repos/NodeBB/NodeBB/tags', {
```
Comment (pitaj) [inline src/admin/versions.js]: What I was suggesting was something like 

    (nconf.get('github_api') || 'https://api.github.com') + '/repos/NodeBB/NodeBB/tags'

Or, preferably, set the default nconf value for `github_api` and just do 

    nconf.get('github_api') + '/repos/NodeBB/NodeBB/tags'

**Comment by a632079** [comment]
OK , use `request((nconf.any('github_api', 'GITHUB_API') || 'https://api.github.com ') + '/repos/NodeBB/NodeBB/tags', {` instead.

**Comment by a632079** [comment]
removed

**Comment by pitaj** [comment]
You still need to sign the CLA

**Pull request closed without merging**